mapviz_plugins/src/odometry_plugin.cpp

```diff
@@ -278,25 +278,70 @@ namespace mapviz_plugins
       PrintInfo("OK");
     }
   }
+  
+
+  void OdometryPlugin::Paint(QPainter* painter, double x, double y, double scale)
+  {
+    int interval = ui_.show_timestamps->value();
+    if (interval == 0)
```

Comment:
"Naked if" statements are dangerous for a number of reasons, primary among which is that a future modification may fail to notice the lack of braces and attempt to add a statement to this block. In this case, either that statement will only execute when the condition is false, which is the exact opposite of what is intended, or the function will *always* return, which is not desired either.

Always use braces to denote blocks, even if they are only one line.